tuteria_application_test/users/models.py

```diff
@@ -5,26 +5,83 @@
 from django.core.urlresolvers import reverse
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import ugettext_lazy as _s
+from django.db.models import Q
 
 
+
+class UserQuerySet(models.QuerySet):
+
+    def attached_bookings(self):
+        return self.annotate(booking_count=models.Count('orders'))
+
+    def with_bookings(self):
+        return self.attached_bookings().filter(booking_count__gt=0)
+
+    def with_transaction_total(self):
+        return self.annotate(transaction_total=models.Sum('wallet__transactions__total')).exclude(transaction_total=None).order_by('-transaction_total')
+
+    def with_transaction_and_booking(self):
+        return self.with_transaction_total().exclude(orders=None)
+
+    def no_bookings(self):
+
+        return self.attached_bookings().filter(booking_count=0).with_transaction_total()
+
+
+class CustomUserManager(UserManager):
+
+    def bookings_aggs(self):
+        print (self.model.id)
+       
+        from django.db.models import IntegerField, Sum
+        return User.objects.all().annotate(completed=models.Count(models.Case(models.When(
```

Comment:
Since you are implementing a model manager method that u would eventually be assigning to the `objects` class  variable. Why call it here instead? Since in the `User` class objects is equal to an instance of this class. So it should mean, I shouldn't be able to access `objects` inside the instance it is assigned to. 

How did you come across this solution?